### PR TITLE
Holodeck update

### PR DESCRIPTION
**Pull request by Lokiamis**
Preview: https://webmshare.com/WJEbK
Pictures: http://imgur.com/a/pcCNM

+Expanded to 11x10 (up from 5x10) on Box and Meta
+All of the older holodeck rooms have been expanded to match new dimensions. 
+Windows are tinted and doors have been switched to prevent most light from bleeding in and give the holodeck privacy. Very immersive if you keep your flashlights off. 
+8 new holodeck rooms have been added
+Wired the radioactive death transmitter machine in Box maint because Kurf told me to
-Ghetto theater on Box had to be cut. What space was left has been reworked. Nothing major from other stations were lost.
-Deff will be a separate PR since Probe is 2slow, so the holodeck will look goofy if this is merged by itself. It won't break the map(I hope), the new rooms just won't fit the holodeck properly.
-There are almost certainly new material farming exploits to be found and fixed. Pretty sure you can't grab anything rare, like wood.

box

meta

:cl: 
 * rscadd: Holodeck updated

### CODE REVIEW

**Comment by Ephx1** [comment]
AI Hologram implementation 2017

**Comment by Lokiamis** [comment]
@Ephx1 law 2 join my basketball team

**Comment by PJB3005** [comment]
Could you shuffle around the walls so that double walls like this are gone? They look really bad in game.

**Comment by ultraplen** [comment]
rip ghetto theatre. I´ve never ever seen it used but it had potential.

**Comment by Lokiamis** [comment]
@PJB3005

**Comment by Ephx1** [comment]
I'm curious as to why you kept the viewing area south of the holodeck on box since it's all tinted windows now.

**Comment by Shadowmech88** [comment]
Once I *eventually* get around to fixing one-way windows that area could be used to see in to the holodeck without allowing people inside the holodeck to see out.

**Comment by Aurx** [comment]
I don't like the airlock sprite that was chosen.  That icon is synonymous with retro-areas like tcommsat, which I don't feel to be a fitting association.  
I think the standard grey with green stripe airlock would fit better, or maybe the stark-white one.

**Comment by Lokiamis** [comment]
@Ephx1 As a hazard and as a window to space, like the 3 above the bridge. While you can't see in, you can still see if there's activity inside by the lighting.
@Shadowmech88 That'd be good. You can see through tinted windows if you're right next to them, but only if there's one on the tile.
@Aurx I'll throw some grey/greens in.

**Comment by CptWad** [comment]
got a few photos of what the new holospaces look like?

**Comment by PJB3005** [comment]
You know we could probably enable lighting on the holdeck. I think the area copying code it uses is awful though and fucks up lighting.

**Comment by Lokiamis** [comment]
Old
old
New
new1

new2

new3

new4



Things I'm not happy with
-The "space" room still looks like shit. Would be nice to get properly animated/scrolling space tiles
-The "meeting room" carpeting is shit. Think it's missing pieces or something because I couldn't get the edges to blend well
-"Theater" room costumes will have to be hand placed until #13669 is fixed. Instruments that spawn in the holodeck don't work for some reason, but instruments you take into the room from elsewhere will. Had to axe the piano for this reason.
-Couldn't get linked machines to work for some reason. Specifically an airlock with a bolt switch in the rage cage. Didn't try a range target device for this reason
-Tajarans in the "zoo" room don't spawn with the abhorrent furry snowflake names I gave them, just show up as Unknown
-Holo windoors would be nice. I tried and failed to add these.

@PJB3005 What's wrong with the lighting?

**Comment by PJB3005** [comment]
@Lokiamis I mean, we should enable dynamic lighting on the holodeck. Right now it's always 100% lit.

**Comment by Lokiamis** [comment]
@PJB3005 You would have to do it on an area-by-area basis unless you want a beach illuminated by florescent bulbs.

**Comment by PJB3005** [comment]
@Lokiamis of course.

File: code/game/machinery/computer/HolodeckControl.dm [hunk 1]
```diff
@@ -572,3 +641,10 @@
 
 	for(var/mob/M in currentarea)
 		to_chat(M, "FIGHT!")
+
+
+
+
+
+
+
```
Comment (N3X15) [inline code/game/machinery/computer/HolodeckControl.dm]: wat

**Review by N3X15 — CHANGES_REQUESTED** [review]
Fix dem lines.

**Review by N3X15 — APPROVED** [review]

**Comment by CptWad** [comment]
Oh no #13734
PROOOOBE
THIS MAP EDIT WAS FIRST
AND APPROVED BEFORE YOURS

**Comment by Lokiamis** [comment]
I can add the missing pipe if nobody cares.

File: code/game/machinery/computer/HolodeckControl.dm [hunk 2]
```diff
@@ -354,12 +406,16 @@
 					var/turf/simulated/floor/FF = get_step(src,direction)
 					FF.update_icon() //so siding get updated properly
 
+/turf/simulated/floor/holofloor/light
+	name = "Light floor"
```
Comment (PJB3005) [inline code/game/machinery/computer/HolodeckControl.dm]: Remove capitalization here.

File: code/game/machinery/computer/HolodeckControl.dm [hunk 3]
```diff
@@ -354,12 +406,16 @@
...
+	luminosity = 5
+	icon_state = "light_on"
+	floor_tile
```
Comment (PJB3005) [inline code/game/machinery/computer/HolodeckControl.dm]: What? This doesn't seem to define anything.

File: code/game/machinery/computer/HolodeckControl.dm [hunk 4]
```diff
@@ -391,6 +447,19 @@
 /obj/structure/window/reinforced/holo/spawnBrokenPieces()
 	return
 
+/obj/structure/window/holo/spawnBrokenPieces()
+	return
+
+/obj/structure/window/holo/attackby(obj/item/weapon/W as obj, mob/user as mob)
```
Comment (PJB3005) [inline code/game/machinery/computer/HolodeckControl.dm]: Remove `as obj` and `as mob` here.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
How do the new areas work on deff which wasn't modified?

File: code/game/machinery/computer/HolodeckControl.dm (hunk 4 above)
Comment (Lokiamis) [reply code/game/machinery/computer/HolodeckControl.dm]: Do you want them gone from the reinforced window that already exists too?

File: code/game/machinery/computer/HolodeckControl.dm (hunk 3 above)
Comment (Lokiamis) [reply code/game/machinery/computer/HolodeckControl.dm]: Eh? I removed these lines and the light floors stopped working. I don't know what you mean.

**Comment by Lokiamis** [comment]
@PJB3005  
>How do the new areas work on deff which wasn't modified?

>Deff will be a separate PR since Probe is 2slow, so the holodeck will look goofy if this is merged by itself. It won't break the map(I hope), the new rooms just won't fit the holodeck properly.

They fill the holodeck tiles that are available starting from the left. It's ugly, but I don't know of a workaround besides waiting. Probe has priority for Deff right now.

File: code/game/machinery/computer/HolodeckControl.dm (hunk 4 above)
Comment (PJB3005) [reply code/game/machinery/computer/HolodeckControl.dm]: Eh leave that.

File: code/game/machinery/computer/HolodeckControl.dm (hunk 3 above)
Comment (PJB3005) [reply code/game/machinery/computer/HolodeckControl.dm]: Well OK then.

**Comment by PJB3005** [comment]
Ah, OK. 

I was worried it might bulldoze the rest of the station or such.

**Review by PJB3005 — APPROVED** [review]

File: code/game/machinery/computer/HolodeckControl.dm (hunk 3 above)
Comment (Intigracy) [reply code/game/machinery/computer/HolodeckControl.dm]: Turf code is stupid weird and all floors have a floor_tile subtype.

**Comment by Yelier** [comment]
Really cool stuff! Did you get the chess idea from the thread? I remember shilling for it a lot.

**Comment by Lokiamis** [comment]
Man, I was probably shilling it with you. All the pieces are labeled, by the way. Some of the constructs look alike.

**Comment by Lokiamis** [comment]
@Probe1 are you fucking serious man #13730

**Comment by Probe1** [comment]
I knew your PR had a needs fixing on it and didn't check it closely enough. Sorry.

**Comment by Zth--** [comment]
These conflicts shouldn't be that hard to fix, they're only a few. But yes, my PR should've been merged last

**Comment by CptWad** [comment]
RIP

**Comment by Probe1** [comment]
Look just restore it and I'll fix the conflicts.

**Comment by Lokiamis** [comment]
It's gone. I'm trying to fix it in another way, which isn't working.

**Comment by CptWad** [comment]
everyone knows that the second map conflict is unfixable. I just hope to see it back again soon.

**Pull request closed without merging**